hemi/grid_stride_loop.h

```diff
@@ -0,0 +1,65 @@
+///////////////////////////////////////////////////////////////////////////////
+// 
+// "Hemi" CUDA Portable C/C++ Utilities
+// 
+// Extended by Brandon Wilson
+//
+// License: BSD License, see LICENSE file in Hemi home directory
+//
+// The home for Hemi is https://github.com/harrism/hemi
+//
+///////////////////////////////////////////////////////////////////////////////
+// Please see the file README.md (https://github.com/harrism/hemi/README.md) 
+// for full documentation and discussion.
+///////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+/////////////////////////////////////////////////////////////////
+// Some utility code to easily loop in a grid-stride pattern on the device, or iteratively on the host
+#include "hemi.h"
+#include "device_api.h"
+
+namespace hemi {
+	
+	///////Grid Stride Loops///////////////////////////////////////////////////
+#define GRID_STRIDE_LOOP(name, start, size) \
```

Comment:
I don't really like adding this unnecessary syntactic sugar in the form of preprocessor macros. I think the range-based for loop syntax is beautifully clear and concise, I wouldn't want to hide the semantics inside a macro just to save some typing. Plus, I already extended the cpp11-range class to support CUDA grids, which you aren't using here.